build-tools/packages/build-cli/src/library/typescriptApi.ts

```diff
@@ -35,29 +35,47 @@ function isTypeExport(_decl: ExportedDeclarations): boolean {
 	return false;
 }
 
+/**
+ * Searches given JSDocs for known {@link ApiLevel} tag.
+ *
+ * @returns Recognized {@link ApiLevel} from JSDocs or undefined.
+ */
+function getLevelFromDocs(jsdocs: JSDoc[]): ApiLevel | undefined {
+	for (const jsdoc of jsdocs) {
+		for (const tag of jsdoc.getTags()) {
+			const tagName = tag.getTagName();
+			if (isKnownApiLevel(tagName)) {
+				return tagName;
+			}
+		}
+	}
+	return undefined;
+}
+
 /**
  * Searches given Node's JSDocs for known {@link ApiLevel} tag.
  *
  * @returns Recognized {@link ApiLevel} from JSDocs or undefined.
  */
 function getNodeLevel(node: Node): ApiLevel | undefined {
 	if (Node.isJSDocable(node)) {
-		for (const jsdoc of node.getJsDocs()) {
-			for (const tag of jsdoc.getTags()) {
-				const tagName = tag.getTagName();
-				if (isKnownApiLevel(tagName)) {
-					return tagName;
-				}
-			}
-		}
-	} else {
-		// Some nodes like `VariableDeclaration`s are not JSDocable, but an ancestor
-		// like `VariableStatement` is and may contain tag.
-		const parent = node.getParent();
-		if (parent !== undefined) {
-			return getNodeLevel(parent);
-		}
+		return getLevelFromDocs(node.getJsDocs());
```

Comment:
It just occurred to me - does this analyze / trim based on the tags of items being re-exported from other packages? That's useful for our own packages, but could be problematic if we happen to re-export from some external library that happens to use TSDoc release tags.

Likely not something we _really_ need to be worried about, but could occur in theory, and is something API-Extractor has to contend with (albeit, not well, currently). Might be worth a note though.

Reply:
This logic should let us retag things, but I didn't really get into the details.
I do know that we have one type that we surface from another library - `EventEmitter`. I know because the rewriter logic complains about it not having a tag when re-exported by example-utils.